.eslintrc.cjs

```diff
@@ -12,17 +12,33 @@ module.exports = {
 	overrides: [
 		{
 			extends: ["plugin:markdown/recommended"],
-			files: ["**/*.{md}"],
+			files: ["**/*.md", "**/*.md/*.js"],
 			processor: "markdown/markdown",
 		},
+		{
+			// https://github.com/eslint/eslint-plugin-markdown/issues/114#issuecomment-818463890
+			files: ["**/*.md/*.ts"],
+			parser: "@typescript-eslint/parser",
+			// Only rules that don't require type information are allowed
+			extends: [
+				"plugin:@typescript-eslint/recommended",
+				"plugin:typescript-sort-keys/recommended",
+			],
+			rules: {
+				"@typescript-eslint/no-unused-vars": "off",
+				"@typescript-eslint/padding-line-between-statements": "off",
+			},
+		},
 		{
 			extends: [
 				"plugin:@typescript-eslint/recommended",
 				"plugin:@typescript-eslint/recommended-requiring-type-checking",
 				"plugin:typescript-sort-keys/recommended",
 				"plugin:@typescript-eslint/strict",
 			],
-			files: ["**/*.{ts,tsx}"],
+			files: ["**/*.ts"],
+			// It's impossible to retrieve type-level information for TypeScript code blocks
```

Comment:
[Nitpick] I think this comment goes without saying - or at least it will if this overrides object is only used for enabling typed rules:

```suggestion
```